soaktest/soakclient_transaction/soakclient_transaction.go

```diff
@@ -0,0 +1,552 @@
+		partitionPosition = append(partitionPosition, v)
+	}
+	return partitionPosition
+}
+
+// terminate closes producer and consumer, and also pass error to the error channel
+func terminate(c *kafka.Consumer, p *kafka.Producer, err error, errorChan chan error) {
+	c.Close()
+	p.Close()
+	errorChan <- err
+}
+
+// verifyProducerTransaction receivev messages from input topic,
+// then produce it the output topic with transaction producer.
+// Commit transaction every 100 messages or every 1 second which every
+// comes first
+func verifyProducerTransaction(broker, inputTopic, outTopic, groupId,
+	transactionId, ccloudAPIKey, ccloudAPISecret string, wg *sync.WaitGroup,
+	termChan chan bool, errorChan chan error) {
+	defer wg.Done()
+	transactionCommitNum := 1
+	hwmarks := make(map[string]uint64)
+	partitionPositionMap := make(map[int32]kafka.TopicPartition)
+	ticker := time.NewTicker(1000 * time.Millisecond)
+
+	c, err := kafka.NewConsumer(&kafka.ConfigMap{
+		"bootstrap.servers":        broker,
+		"group.id":                 groupId,
+		"go.events.channel.enable": true,
+		"enable.auto.commit":       false,
+		"statistics.interval.ms":   5000,
+		"sasl.mechanisms":          "PLAIN",
+		"security.protocol":        "SASL_SSL",
+		"sasl.username":            ccloudAPIKey,
+		"sasl.password":            ccloudAPISecret})
+
+	if err != nil {
+		fmt.Printf("Failed to create consumer: %s\n", err)
+		errorChan <- err
+		return
+	}
+
+	err = c.Subscribe(inputTopic, nil)
+	if err != nil {
+		fmt.Printf("Failed to subscribe to topic: %s with error %s\n",
+			inputTopic, err)
+		c.Close()
+		errorChan <- err
+		return
+	}
+
+	p, err := kafka.NewProducer(&kafka.ConfigMap{
+		"bootstrap.servers":      broker,
+		"transactional.id":       transactionId,
+		"statistics.interval.ms": 5000,
+		"sasl.mechanisms":        "PLAIN",
+		"security.protocol":      "SASL_SSL",
+		"sasl.username":          ccloudAPIKey,
+		"sasl.password":          ccloudAPISecret,
+	})
+	if err != nil {
+		fmt.Errorf("Failed to create transactional producer: %s", err)
+		c.Close()
+		errorChan <- err
+		return
+	}
+
+	maxDuration, err := time.ParseDuration("15s")
+	if err != nil {
+		fmt.Printf("Failed to parse Duration %v", err)
+		terminate(c, p, err, errorChan)
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), maxDuration)
+	defer cancel()
+
+	// If InitTransactions succeed, will continue to BeginTransaction
+	// If InitTransactions failed with err.(kafka.Error).IsRetriable(),
+	// sleep SLEEP_TIME then retry
+	// If failed with other errors, return the fatal error
+	for i := 0; i < RETRY_NUM; i++ {
+		err = p.InitTransactions(ctx)
+		if err != nil {
+			fmt.Errorf("InitTransactions() failed: %s\n", err)
+			if err.(kafka.Error).IsRetriable() {
+				if i == RETRY_NUM-1 {
+					fmt.Printf("InitTransactions() failed after %d "+
+						"times retries: %s\n", RETRY_NUM, err)
+					terminate(c, p, err, errorChan)
+					return
+				}
+				time.Sleep(SLEEP_TIME)
```

Comment:
I find this naming harder to comprehend than actual values, "SLEEP_TIME" is ambigious/generic and when seen at the definition it doesn't say what the sleep time is used for, and when seen here it doesn't give us an idea how long this sleeps.
Either have a INIT_RETRY_SLEEP_TIME, or just specify the actual number here (I would go with the latter.. less boilerplate)